src/components/FingerStatistics.tsx

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import Box from "@mui/material/Box";
+
+type Hand = "left" | "right";
+
+type Finger = "thumb" | "first" | "second" | "third" | "fourth";
+
+type KeyData = {
+  keyName: string;
+  hand: Hand;
+  finger: Finger;
+  pushCount: number;
+  missCount: number;
+  timeSecCount: number;
+};
+
+type Props = {
+  data: KeyData[];
+};
+
+const FingerStatistics = (props: Props) => {
+  const { data } = props;
+  // キー毎の情報を指ごとの情報に変換する
+  // 後述のObject.keys()後のforEach()で型エラーにならないようにHandとFingerにstring型を追加しておく
+  const fingerData: {
+    [key1 in Hand | string]: {
+      [key2 in Finger | string]: { pushCountSum: number; missCountSum: number; timeSecSum: number };
+    };
+  } = {
+    left: {
+      thumb: { pushCountSum: 0, missCountSum: 0, timeSecSum: 0 },
+      first: { pushCountSum: 0, missCountSum: 0, timeSecSum: 0 },
+      second: { pushCountSum: 0, missCountSum: 0, timeSecSum: 0 },
+      third: { pushCountSum: 0, missCountSum: 0, timeSecSum: 0 },
+      fourth: { pushCountSum: 0, missCountSum: 0, timeSecSum: 0 },
+    },
+    right: {
+      thumb: { pushCountSum: 0, missCountSum: 0, timeSecSum: 0 },
+      first: { pushCountSum: 0, missCountSum: 0, timeSecSum: 0 },
+      second: { pushCountSum: 0, missCountSum: 0, timeSecSum: 0 },
+      third: { pushCountSum: 0, missCountSum: 0, timeSecSum: 0 },
+      fourth: { pushCountSum: 0, missCountSum: 0, timeSecSum: 0 },
+    },
+  };
+  data.forEach((element) => {
+    fingerData[element.hand][element.finger].pushCountSum += element.pushCount;
+    fingerData[element.hand][element.finger].missCountSum += element.missCount;
+    fingerData[element.hand][element.finger].timeSecSum += element.timeSecCount;
+  });
+  // 指ごとの統計情報を表示する
+  // 指ごとの統計情報に親指は不要なので各手の指配列をslice(1)する
+  const statistics = Object.keys(fingerData).map((hand) => {
+    const eachHandResults = Object.keys(fingerData[hand])
```

Comment:
```suggestion
  const statistics = (Object.keys(fingerData) as Hand[]).map((hand) => {
    const eachHandResults = (Object.keys(fingerData[hand]) as Finger[])
```
型アサーションといって、`as` を使用すると、返り値をその場で任意の型に指定することができます。
これはこれで裏技っぽいのですが、string型を許容するよりかはいいのでこちらにしましょう！

Reply:
型アサーションを使って型の厳密性を維持する方法、勉強になりました！ありがとうございます。
そのように訂正します！